### PR TITLE
[feature/ASV-1687] Errors in Search

### PR DESCRIPTION
**Pull request by franciscoaleixo**
**What does this PR do?**

   This PR refactors the whole Error view so that we reuse the same logic and the same view everywhere. It also adds error support to search.

**Database changed?**

   No

**How should this be manually tested?**

  Every error should be checked, as it was changed, but especially the new errors for search that were added. You can check this by disabling internet and searching. Then re-enable and click on retry. I'm not sure how you could test generic errors though.

**What are the relevant tickets?**

  [ASV-1687](https://aptoide.atlassian.net/browse/ASV-1687)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/view/swipe/LoaderLayoutHandler.java [hunk 1]
```diff
@@ -30,10 +31,7 @@
   private final List<Integer> viewsToShowAfterLoadingId = new ArrayList<>();
   protected ProgressBar progressBar;
   private List<View> viewsToShowAfterLoading = new ArrayList<>();
-  private View genericErrorView;
-  private View noNetworkConnectionView;
-  private View retryErrorView;
-  private View retryNoNetworkView;
+  private ErrorView errorView;
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/view/swipe/LoaderLayoutHandler.java]: Not being unbinded. I know that the other views are not being unbinded as well and you are not the one responsible for this, but do you know why ?

File: app/src/main/res/layout/recycler_swipe_fragment.xml [hunk 2]
```diff
@@ -7,11 +7,18 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/recycler_swipe_fragment.xml]: codestyle

File: app/src/main/res/layout/fragment_editorial_list.xml [hunk 3]
```diff
@@ -3,6 +3,7 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/fragment_editorial_list.xml]: codestyle

File: app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -3,6 +3,7 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/fragment_home.xml]: codestyle

File: app/src/main/res/layout/my_stores.xml [hunk 5]
```diff
@@ -3,6 +3,7 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/my_stores.xml]: codestyle

File: app/src/main/res/layout/recycler_swipe_fragment_with_toolbar.xml [hunk 6]
```diff
@@ -8,6 +8,7 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/recycler_swipe_fragment_with_toolbar.xml]: codestyle

File: app/src/main/res/layout/social_fragment_layout.xml [hunk 7]
```diff
@@ -3,6 +3,7 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/social_fragment_layout.xml]: codestyle

File: app/src/main/res/layout/store_activity.xml [hunk 8]
```diff
@@ -9,22 +9,17 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (phillipsgoncalves) [inline app/src/main/res/layout/store_activity.xml]: code style

File: aptoide-views/src/main/res/layout/error_view.xml [hunk 9]
```diff
@@ -1,18 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (phillipsgoncalves) [inline aptoide-views/src/main/res/layout/error_view.xml]: codestyle

File: app/src/main/java/cm/aptoide/pt/view/swipe/LoaderLayoutHandler.java (hunk 1 above)
Comment (franciscoaleixo) [reply app/src/main/java/cm/aptoide/pt/view/swipe/LoaderLayoutHandler.java]: Only God knows 🙏 

But seriously, I don't, someone probably forgot.

**Pull request merged**